internal/scheduler/scheduler.go

```diff
@@ -195,70 +209,96 @@ func (s *Scheduler) start() {
 
 func (s *Scheduler) runJob(r *job.Run) error {
 	jobId := JobId(r.JobId)
-	j, ok := s.registeredJobs.Load(jobId)
-	job := j.(Job)
-	s.logger.Debug(fmt.Sprintf("starting job run %q for job (%v: %v)", r.PrivateId, jobId, job.Name()))
+	regJob, ok := s.registeredJobs.Load(jobId)
+	j := regJob.(Job)
+	s.logger.Debug("starting job run", "run id", r.PrivateId, "job id", jobId, "name", j.Name())
 	if !ok {
-		return fmt.Errorf("job (%v: %v) not registered on scheduler", jobId, job.Name())
+		return fmt.Errorf("job (%v: %v) not registered on scheduler", jobId, j.Name())
 	}
 
 	repo, err := s.jobRepoFn()
 	if err != nil {
 		return fmt.Errorf("failed to create job repository: %w", err)
 	}
 
-	s.l.Lock()
-	if _, ok := s.runningJobs[jobId]; ok {
-		s.l.Unlock()
-		return fmt.Errorf("job (%v: %v) is already running", jobId, job.Name())
-	}
 	jobContext, jobCancel := context.WithCancel(s.baseContext)
-	s.runningJobs[jobId] = runningJob{runId: r.PrivateId, cancelCtx: jobCancel}
-	s.l.Unlock()
+	_, loaded := s.runningJobs.LoadOrStore(jobId, runningJob{runId: r.PrivateId, cancelCtx: jobCancel, status: j.Status})
+	if loaded {
```

Comment:
Should jobCancel be called here?

Reply:
Good question, I was not sure if I wanted to cancel the running job at this point.  The only reason this would happen is if the job has been finished (failed, completed or interrupted)  So in theory the only way we can get here is a race or another server interrupted this.  So perhaps it should be cancelled yes!